[Test] org.eclipse.text.quicksearch.tests.QuickAccessComputerTest.testQuickAccessComputer fails randomly

Failed randomly on **Mac** for #3004 : https://github.com/eclipse-platform/eclipse.platform.ui/actions/runs/15628923632/job/44028715479?pr=3004#step:5:14405

```
org.eclipse.text.quicksearch.tests.QuickAccessComputerTest.testQuickAccessComputer -- Time elapsed: 0.399 s <<< FAILURE!
java.lang.AssertionError: expected:<1> but was:<0>
	at org.junit.Assert.fail(Assert.java:89)
	at org.junit.Assert.failNotEquals(Assert.java:835)
	at org.junit.Assert.assertEquals(Assert.java:647)
	at org.junit.Assert.assertEquals(Assert.java:633)
	at org.eclipse.text.quicksearch.tests.QuickAccessComputerTest.testQuickAccessComputer(QuickAccessComputerTest.java:60)
```

--- a/tests/org.eclipse.text.quicksearch.tests/src/org/eclipse/text/quicksearch/tests/QuickAccessComputerTest.java
+++ b/tests/org.eclipse.text.quicksearch.tests/src/org/eclipse/text/quicksearch/tests/QuickAccessComputerTest.java
@@ -9,7 +9,6 @@
  *******************************************************************************/
 package org.eclipse.text.quicksearch.tests;
 
-import static org.junit.jupiter.api.Assertions.assertEquals;
 import static org.junit.jupiter.api.Assertions.assertTrue;
 
 import java.io.IOException;
@@ -55,8 +54,16 @@ class QuickAccessComputerTest {
 		dialog.setInitialPattern(request);
 		dialog.setBlockOnOpen(false);
 		dialog.open();
-		assertTrue(DisplayHelper.waitForCondition(dialog.getShell().getDisplay(), 2000, () -> dialog.getResult().length > 0));
+		var display = dialog.getShell().getDisplay();
+		assertTrue(DisplayHelper.waitForCondition(display, 2000, () -> dialog.getResult().length > 0));
 		dialog.close();
-		assertEquals(1, new QuickSearchQuickAccessComputer().computeElements(request, new NullProgressMonitor()).length);
+
+		// Wait for the QuickAccessComputer to return results, as the search happens asynchronously.
+		// Retry the search if it initially returns no results, allowing time for the background
+		// search job to find matches. This addresses race conditions on slower CI systems.
+		assertTrue(DisplayHelper.waitForCondition(display, 5000, () -> {
+			QuickSearchQuickAccessComputer computer = new QuickSearchQuickAccessComputer();
+			return computer.computeElements(request, new NullProgressMonitor()).length == 1;
+		}), "Expected QuickAccessComputer to find exactly one result within timeout");
 	}
 }

